pkg/actor/cluster_restart_test.go

```diff
@@ -0,0 +1,215 @@
+				Status: metav1.ConditionTrue,
+				LastTransitionTime: metav1.Now(),
+			},
+
+		}
+		require.True(t, cra.Handles(conditions))
+
+	})
+
+	t.Run("Handle func does not pass", func(t *testing.T) {
+
+		// Checked version is false
+		conditions = []api.ClusterCondition{
+			{
+				Type: api.InitializedCondition,
+				Status: metav1.ConditionTrue,
+				LastTransitionTime: metav1.Now(),
+			},
+			{
+				Type: api.CrdbVersionChecked,
+				Status: metav1.ConditionFalse,
+				LastTransitionTime: metav1.Now(),
+			},
+
+		}
+		require.False(t, cra.Handles(conditions))
+
+		// Initialized condition is unknown or missing
+		conditions = []api.ClusterCondition{
+			{
+				Type: api.InitializedCondition,
+				Status: metav1.ConditionUnknown,
+				LastTransitionTime: metav1.Now(),
+			},
+			{
+				Type: api.CrdbVersionChecked,
+				Status: metav1.ConditionTrue,
+				LastTransitionTime: metav1.Now(),
+			},
+
+		}
+		require.False(t, cra.Handles(conditions))
+
+		conditions = []api.ClusterCondition{
+			{
+				Type: api.CrdbVersionChecked,
+				Status: metav1.ConditionTrue,
+				LastTransitionTime: metav1.Now(),
+			},
+
+		}
+		require.False(t, cra.Handles(conditions))
+	})
+
+}
+
+func TestClusterReadyForRestart(t *testing.T) {
+	ssStatus := appsv1.StatefulSetStatus{
+		Replicas: 2,
+		CurrentReplicas: 2,
+	}
+	require.NoError(t, statefulSetReplicasAvailable(&ssStatus))
+
+	ssStatus.Replicas = 0
+	require.Error(t, statefulSetReplicasAvailable(&ssStatus))
+}
+
+func TestFullClusterRestart(t *testing.T) {
+	// Setup fake client
+	builder := fake.NewClientBuilder()
+
+	client := builder.Build()
+
+	cr := newClusterRestart(nil, client, nil).(*clusterRestart)
+	require.NotNil(t, cr)
+	var stsReplicas int32
+	stsReplicas = 3
+	cltSet := fakeclient.NewSimpleClientset()
+
+	sts := createStatefulSet(stsReplicas)
+	cltSet.Tracker().Add(&sts)
+
+	addPodsToStatefulSet(stsReplicas, sts, cltSet)
+
+	cltSet.PrependReactor("*", "*", func(action clienttesting.Action) (handled bool, ret runtime.Object, err error) {
+		tracker := cltSet.Tracker()
+		gvr := action.GetResource()
+		ns := action.GetNamespace()
+		verb := action.GetVerb()
+
+
+		switch verb {
+		case "update":
+			updateAction := action.(clienttesting.UpdateAction)
+			obj := updateAction.GetObject().(*appsv1.StatefulSet)
+			tracker.Update(gvr, obj, ns)
+			return true, obj, nil
+		case "get":
+			getAction := action.(clienttesting.GetAction)
+			obj, err := tracker.Get(gvr, ns, getAction.GetName())
+			if err != nil {
+				return false, nil, err
+			}
+			return true, obj, nil
+		}
+		
+		return false, nil, nil
+	})
+
+	sts.Status.Replicas = stsReplicas
+	sts.Status.ReadyReplicas = stsReplicas
+
+	cltSet.Tracker().Update(schema.GroupVersionResource{
+		Group:    "apps",
+		Version:  "v1",
+		Resource: "statefulset",
+	}, &sts, sts.Namespace)
+	require.NoError(t, cr.fullClusterRestart(context.TODO(), &sts, Log, cltSet))
+}
+
+func createStatefulSet(stsReplicas int32) appsv1.StatefulSet {
```

Comment:
We already have a builder, we should reuse it if we cab

Reply:
The builder is tied to `e2e` or at least was, can we do that in a follow up PR?

Reply:
You want to use the builder for building something other than the CRDB Cluster CRD?